### PR TITLE
Format enums

### PR DESCRIPTION
**Pull request by munificent**
Format enum declarations with and without members.

Enums are a little tricky because they are sort of like lists (i.e. ListPiece) and sort of like type declarations (i.e. BlockPiece). Enums without members should be formatted pretty much like any other delimited list where the declaration can collapse to one line (and discard any trailing comma) if it fits or expand to one element per line (and add a trailing comma) otherwise.

Enums with members are formatted pretty much like blocks where each constant  or member is on its own line.

There are a few wrinkles around trailing commas and the semicolon between the constants and members:

-   If the enum has no members, any trailing `;` is discarded. It doesn't do anything useful and I consider this morally equivalent to discarding other trailing commas.

-   If the enum has members, then a trailing comma after the last constant is redundant with the subsequent `;`. In that case, it discards the comma.

In both of these cases, we have to make sure we don't accidentally lose any comments around that discarded punctuation. So there are a lot of tests for all of those edge cases even though in practice no one writes code like that.

Discarding `;` on enums without members means that we treat `;` like whitespace, so I also loosened equalIgnoringWhitespace() to ignore semicolons too.

Since enums with and without members are formatted using different codepaths, there are a lot of mostly redundant-seeming tests for them.

### CODE REVIEW

**Comment by natebosch** [comment]
> I also loosened equalIgnoringWhitespace() to ignore semicolons too.

Does this reduce our ability to catch a missed `token(semicolon);`? Should we make it an optional behavior and only use the looser check when we need it?

**Review by natebosch — APPROVED** [review]

File: lib/src/front_end/ast_node_visitor.dart [hunk 1]
```diff
@@ -400,12 +403,81 @@ class AstNodeVisitor extends ThrowingAstVisitor<void>
...
+
+      var sequence = SequenceBuilder(this);
+      for (var constant in node.constants) {
+        sequence.addCommentsBefore(constant.firstNonCommentToken);
+        visit(constant);
+
+        if (constant != node.constants.last) {
+          commaAfter(constant);
+        } else {
+          // Discard the trailing comma if there is one since there is a
```
Comment (kallentu) [inline lib/src/front_end/ast_node_visitor.dart:449]: Yes! Not having `,;` haha

**Review by kallentu — APPROVED** [review]
